tools/kagami/src/genesis.rs

```diff
@@ -241,10 +259,251 @@ fn generate_synthetic(
+        let key_pair = if let Some(path) = self.keypair_file {
+            read_keypair(path)?
+        } else {
+            let (public_key, private_key) = (self.get_public_key()?, self.get_private_key()?);
+            KeyPair::new(public_key, private_key)?
+        };
+
+        let genesis_block = RawGenesisBlock::from_path(&self.genesis_file)?;
+        let genesis_signature =
+            GenesisNetwork::new_genesis_signature(genesis_block, &self.chain_id, &key_pair);
+        let genesis_signature_string = genesis_signature.to_hex_string();
+
+        let encoded_genesis_signature = if self.scale {
+            genesis_signature_string.encode()
+        } else {
+            Vec::default()
+        };
+
+        let hex_genesis_config = if self.hex {
+            genesis_signature_string
+        } else {
+            String::default()
+        };
+
+        if let Some(path) = self.out_file {
+            if self.scale {
+                fs::write(&path, encoded_genesis_signature)?;
+            } else {
+                fs::write(&path, hex_genesis_config)?;
+            }
+
+            writeln!(
+                writer,
+                "Genesis was successfully signed and written to `{}`",
+                path.display()
+            )?;
+        } else if self.scale {
+            writeln!(
+                writer,
+                "SCALE encoded data is not supported for console outputs."
+            )?;
+        } else {
+            writeln!(writer, "{hex_genesis_config}")?;
+        }
+
+        Ok(())
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use iroha_genesis::{GenesisSignature, GenesisTransaction};
+
+    use super::*;
+    use crate::crypto::AlgorithmArg;
+
+    const GENESIS_JSON_PATH: &str = "../../configs/swarm/genesis.json";
+
+    fn genesis_signing_works() -> Result<bool> {
+        let keypair_config = crypto::Args {
+            algorithm: AlgorithmArg::default(),
+            private_key: None,
+            seed: None,
+            json: true,
+            compact: false,
+        };
+
+        let mut keypair_json = BufWriter::new(Vec::new());
+        keypair_config.run(&mut keypair_json)?;
+        let keypair: KeyPair = serde_json::from_slice(keypair_json.buffer())?;
+
+        let tmp_keypair_json_file = tempfile::NamedTempFile::new()?;
+        fs::write(tmp_keypair_json_file.path(), keypair_json.buffer())?;
+
+        let chain_id = ChainId::from("0123456");
+        let crypto_genesis_config = SignArgs {
+            algorithm: AlgorithmArg::default(),
+            private_key_string: None,
+            private_key_file: None,
+            public_key_string: None,
+            public_key_file: None,
+            keypair_file: Some(PathBuf::from(tmp_keypair_json_file.path())),
+            chain_id: chain_id.clone(),
+            genesis_file: PathBuf::from_str(GENESIS_JSON_PATH)?,
+            out_file: None,
+            scale: false,
+            hex: true,
+        };
+
```

Comment:
Would be cool to construct `SignArgs` from raw CLI args, in a way I did it here:

https://github.com/hyperledger/iroha/blob/ecd311983a6af96dd4c4c5ca65acc0e89d2d1b0d/cli/src/lib.rs#L997-L1002